packages/fhir/test/Adaptor.test.js

```diff
@@ -130,56 +159,99 @@ describe('get', () => {
   });
 
   it('should get patient resource by id', async () => {
-    const state = {
-      configuration: {
-        baseUrl: 'https://hapi.fhir.org',
-        apiPath: 'baseR4',
-      },
-    };
+    testServer
+      .intercept({
+        path: '/baseR4/Patient/abc',
+        method: 'GET',
+      })
+      .reply(200, fixtures.patient)
+      .persist();
+    const state = { configuration };
+
+    const path1State = await execute(get('/Patient/abc'))(state);
+    const path2State = await execute(get('Patient/abc'))(state);
+
+    expect(path1State.data).to.eql(fixtures.patient);
+    expect(path2State.data).to.eql(fixtures.patient);
+  });
+});
+
+describe('getClaim', () => {
+  it('should get claim resource bundle', async () => {
+    testServer
+      .intercept({
+        path: '/baseR4/Claim',
+        method: 'GET',
+      })
+      .reply(200, fixtures.claimBundle);
+
+    const state = { configuration };
 
-    const finalState = await execute(get('Patient/592442'))(state);
+    const finalState = await execute(getClaim())(state);
 
-    expect(finalState.data).to.eql(fixtures.patient);
+    expect(finalState.data).to.eql(fixtures.claimBundle);
   });
 
-  it('should throw for invalid patient id', async () => {
+  it('should get claim by id', async () => {
+    testServer
+      .intercept({
+        path: '/baseR4/Claim/49023',
+        method: 'GET',
+      })
+      .reply(200, fixtures.claim);
+
     const state = {
-      configuration: {
-        baseUrl: 'https://hapi.fhir.org',
-        apiPath: 'baseR4',
-      },
+      configuration,
     };
 
-    const finalState = await execute(get('Patient/invalid-patient-id'))(state);
+    const finalState = await execute(getClaim('49023'))(state);
 
-    expect(finalState.data).to.eql(fixtures.invalidPatient);
+    expect(finalState.data).to.eql(fixtures.claim);
   });
 });
 
-describe('getClaim', () => {
-  it('should get claim resource bundle', async () => {
+describe('post', () => {
+  it('throws an error for a 404', async () => {
+    testServer
+      .intercept({
+        path: 'baseR4/noAccess',
+        method: 'POST',
+      })
+      .reply(404, fixtures.noAccessResponse);
+
     const state = {
-      configuration: {
-        baseUrl: 'https://hapi.fhir.org',
-        apiPath: 'baseR4',
-      },
+      configuration,
     };
 
-    const finalState = await execute(getClaim())(state);
+    let e;
+    try {
+      await execute(post('noAccess', { name: 'taylor' }))(state);
+    } catch (err) {
+      e = err;
+    }
 
-    expect(finalState.data).to.eql(fixtures.claimBundle);
+    expect(e.message).to.include(
+      'POST to https://hapi.fhir.org/baseR4/noAccess returned 404: Not Found'
+    );
+    expect(e.statusCode).to.eql(404);
+    expect(e.statusMessage).to.eql('Not Found');
   });
+});
 
-  it('should get claim by id', async () => {
+describe('Url Matches BaseUrl', () => {
```

Comment:
This shouldn't be in its own describe block, unless it's going to test the function in isolation.

Just strip the describe out and  tuck the test under the post tests

Reply:
Done ✅